dot/sync/chain_processor.go

```diff
@@ -249,9 +259,7 @@ func (s *chainProcessor) handleBlock(block *types.Block) error {
 }
 
 func (s *chainProcessor) handleJustification(header *types.Header, justification []byte) (err error) {
-	if len(justification) == 0 {
-		return nil
-	}
```

Comment:
Here if the justification isn't `nil`, I think we should process it even if it's empty right?? That looks like a potential attack :thinking:

Reply:
length justification check is pushed up the call stack, although it's likely unnecessary since the blockdata would have it set to `nil` instead of the empty byte slice.